Adds "box-sizing: border-box" to table in UA stylesheet

karlcow opened the pull request:
#### 8915be4c047012129bacf9c755fc56e925a397e3
<pre>
Adds &quot;box-sizing: border-box&quot; to table in UA stylesheet
<a href="https://bugs.webkit.org/show_bug.cgi?id=243234">https://bugs.webkit.org/show_bug.cgi?id=243234</a>
&lt;rdar://97634949&gt;

Reviewed by Tim Nguyen.

This changes the UA stylesheet to match to the spec
and adjusts the relevant tests.
<a href="https://html.spec.whatwg.org/#tables-2">https://html.spec.whatwg.org/#tables-2</a>

* Source/WebCore/css/html.css:
(table):
* LayoutTests/fast/borders/border-image-fill-no-border-expected.html:
* LayoutTests/fast/borders/resources/svg-50x50-intrinsic.svg: Added.
* LayoutTests/fast/table/min-width-html-block-table-expected.txt:
* LayoutTests/fast/table/min-width-html-inline-table-expected.txt:
* LayoutTests/fast/table/resources/min-width-helpers.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/position-absolute-table-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/form-controls/resets.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/tables/table-ua-stylesheet-expected.txt:
* LayoutTests/platform/glib/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/glib/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/glib/tables/mozilla/bugs/bug145572-expected.txt:
* LayoutTests/platform/gtk/tables/mozilla_expected_failures/bugs/bug3166-14-expected.png:
* LayoutTests/platform/gtk/tables/mozilla_expected_failures/bugs/bug3166-17-expected.png:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug145572-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/form-controls/resets-expected.txt:
* LayoutTests/platform/mac/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/mac/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug145572-expected.txt:
* LayoutTests/platform/win/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/win/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/win/tables/mozilla/bugs/bug145572-expected.txt:

Canonical link: <a href="https://commits.webkit.org/253581@main">https://commits.webkit.org/253581@main</a>
</pre>










<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ec63def70f0d9135873e760777a9fc078986b9c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/86456 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/30376 "Built successfully") | [✅ ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/17423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95301 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149012 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/90444 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/28777 "Built successfully") | [✅ ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25359 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/78609 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/90545 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92072 "Passed tests") | [✅ ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/23352 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/73425 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23401 "Passed tests") | 
| | [✅ ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/78360 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/17423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66405 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/26688 "Built successfully") | [✅ ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/17423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/26602 "Built successfully") | [✅ ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/17423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2548 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28281 "Built successfully") | [✅ ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/73425 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28222 "Built successfully") | [✅ ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/17423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/form-controls/resets.html:
```diff
@@ -59,6 +59,7 @@
 td, th { display: table-cell; }
 table {
   text-indent: initial;
+  box-sizing: border-box;
```
This is already in above rule, in: `table { display: table; box-sizing: border-box; }`

Is it really needed?

karlcow replied:
not needed. Indeed. I can remove it and push again.

nt1m reviewed (approved):
r=me pending EWS results

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/form-controls/resets.html:
```diff
@@ -57,9 +57,7 @@
-table {
-  text-indent: initial;
-}
+table { text-indent: initial; }
```
nit: please undo this change completely since it's imported (and unlist this file from the commit message as well)

karlcow replied:
huh. Why did I do that. Ah… yes I removed the border-box I had added in there. Sorry about that.

karlcow committed:
Adds "box-sizing: border-box" to table in UA stylesheet
https://bugs.webkit.org/show_bug.cgi?id=243234
<rdar://97634949>

Reviewed by Tim Nguyen.

This changes the UA stylesheet to match to the spec
and adjusts the relevant tests.
https://html.spec.whatwg.org/#tables-2

* Source/WebCore/css/html.css:
(table):
* LayoutTests/fast/borders/border-image-fill-no-border-expected.html:
* LayoutTests/fast/borders/resources/svg-50x50-intrinsic.svg: Added.
* LayoutTests/fast/table/min-width-html-block-table-expected.txt:
* LayoutTests/fast/table/min-width-html-inline-table-expected.txt:
* LayoutTests/fast/table/resources/min-width-helpers.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/position-absolute-table-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/form-controls/resets.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/tables/table-ua-stylesheet-expected.txt:
* LayoutTests/platform/glib/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/glib/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/glib/tables/mozilla/bugs/bug145572-expected.txt:
* LayoutTests/platform/gtk/tables/mozilla_expected_failures/bugs/bug3166-14-expected.png:
* LayoutTests/platform/gtk/tables/mozilla_expected_failures/bugs/bug3166-17-expected.png:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug145572-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/form-controls/resets-expected.txt:
* LayoutTests/platform/mac/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/mac/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug145572-expected.txt:
* LayoutTests/platform/win/tables/mozilla_expected_failures/bugs/bug3166-14-expected.txt:
* LayoutTests/platform/win/tables/mozilla_expected_failures/bugs/bug3166-17-expected.txt:
* LayoutTests/platform/win/tables/mozilla/bugs/bug145572-expected.txt:

Canonical link: https://commits.webkit.org/253581@main
